src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -169,6 +177,11 @@ internal override bool IsDeclarationWithSharedBody(SyntaxNode declaration)
 
         protected override ImmutableArray<ISymbol> GetCapturedVariables(SemanticModel model, SyntaxNode memberBody)
         {
+            if (memberBody is CompilationUnitSyntax unit && unit.ContainsTopLevelStatements())
+            {
+                return model.AnalyzeDataFlow(((GlobalStatementSyntax)unit.Members[0]).Statement, unit.Members.OfType<GlobalStatementSyntax>().Last().Statement)!.Captured;
```

Comment:
>(GlobalStatementSyntax)unit.Members[0] [](http://example.com/codeflow?start=46&length=38)

Might the first member be a type or namespace declaration?

Reply:
Not if `unit.ContainsTopLevelStatements` returns true. Global statements must come before any type declarations, or its a syntax error.